src/components/theme-provider.tsx

```diff
@@ -28,9 +29,8 @@ export function ThemeProvider({
   storageKey = "vite-ui-theme",
   ...props
 }: ThemeProviderProps) {
-  const [theme, setTheme] = useState<Theme>(
-    () => (localStorage.getItem(storageKey) as Theme) || defaultTheme
-  );
+  
```

Comment:
This makes sense, since lines 31-33 were causing the error.

Reply:
Yea we don't need it anymore since I'm using the custom hook now on line 33